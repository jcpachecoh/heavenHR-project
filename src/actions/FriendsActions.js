import * as types from '../constants/ActionTypes';

export function addFriend(name, sex) {
  console.log("dentro funcion"+sex);
  return {
    type: types.ADD_FRIEND,
    name, 
    sex
  };
}

export function deleteFriend(id) {
  console.log(id);
  return {
    type: types.DELETE_FRIEND,
    id
  };
}

export function starFriend(id) {
  return {
    type: types.STAR_FRIEND,
    id
  };
}
