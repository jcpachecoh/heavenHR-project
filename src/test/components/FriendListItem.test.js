import React from 'react';
import FriendListItem from '../../components/FriendListItem';
import * as TestUtils from 'react-addons-test-utils';
import {addFriend, deleteFriend, starFriend} from '../../actions/FriendsActions';

describe('Testing <FriendListItem /> component', () => {
    it('Should render FriendListItem component ', () => {
        const friendsMockData = {
            friendsById: [
              {
                key: 0,
                id: 0,
                name: 'Theodore Roosevelt',
                sex: 'Male',
                starred: true
              },
              {
                key: 1,
                id: 1,
                name: 'Abraham Lincoln',
                sex: 'Male',
                starred: false
              },
              { 
                key: 2,
                id: 2,
                name: 'George Washington',
                sex: 'Female',
                starred: false
              }
            ]
          };

        const actions = {
            addFriend: addFriend,
            deleteFriend: deleteFriend,
            starFriend: starFriend
        }; 
        expect(TestUtils.createRenderer().render(
            <FriendListItem  key={friendsMockData.friendsById[0].key}
            id={friendsMockData.friendsById[0].id}
            name={friendsMockData.friendsById[0].name}
            sex={friendsMockData.friendsById[0].sex}
            starred={friendsMockData.friendsById[0].starred}
            {...actions}  />
          )).toMatchSnapshot()
    });
});
