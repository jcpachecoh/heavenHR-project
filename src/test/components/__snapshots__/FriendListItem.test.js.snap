// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <FriendListItem /> component Should render FriendListItem component  1`] = `
<li
  className={undefined}
>
  <div
    className={undefined}
  >
    <div>
      <span>
        Theodore Roosevelt
      </span>
    </div>
    <div>
      <b>
        Sex: 
      </b>
      Male
    </div>
    <div>
      <small>
        xx friends in common
      </small>
    </div>
  </div>
  <div
    className={undefined}
  >
    <button
      className="btn btn-default undefined"
      onClick={[Function]}
    >
      <i
        className="fa fa-star"
      />
    </button>
    <button
      className="btn btn-default undefined"
      onClick={[Function]}
    >
      <i
        className="fa fa-trash"
      />
    </button>
  </div>
</li>
`;
